plugins/modules/hcloud_iso_info.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2022, Patrice Le Guyader
+# heavily inspired by the work of @LKaemmerling
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+DOCUMENTATION = """
+---
+module: hcloud_iso_info
+
+short_description: Gather infos about the Hetzner Cloud ISO list.
+
+description:
+    - Gather infos about the Hetzner Cloud ISO list.
+
+author:
+    - Patrice Le Guyader (@patlegu)
+    - Lukas Kaemmerling (@LKaemmerling)
+
+options:
+    id:
+        description:
+            - The ID of the ISO image you want to get.
+        type: int
+    name:
+        description:
+            - The description of the ISO you want to get.
+        type: str
+    architecture:
+        description:
+            - Filter ISOs with compatible architecture.
+        type: str
+        choices: [x86, arm]
+    include_architecture_wildcard:
+        description:
+            - Include ISOs with wildcard architecture (architecture is null).
+            - Works only if architecture filter is specified.
+        type: bool
+
+extends_documentation_fragment:
+    - hetzner.hcloud.hcloud
+"""
+
+EXAMPLES = """
+- name: Gather hcloud ISO type infos
+  hcloud_iso_info:
+  register: output
+
+- name: Print the gathered infos
+  debug:
+    var: output.hcloud_iso_info
+"""
+
+RETURN = """
+hcloud_iso_info:
+    description: The ISO type infos as list
+    returned: always
+    type: complex
+    contains:
+        id:
+            description: ID of the ISO
+            returned: always
+            type: int
+            sample: 22110
+        name:
+            description: Unique identifier of the ISO. Only set for public ISOs
+            returned: always
+            type: str
+            sample: debian-12.0.0-amd64-netinst.iso
+        description:
+            description: Description of the ISO
+            returned: always
+            type: str
+            sample: Debian 12.0 (amd64/netinstall)
+        architecture:
+            description: >
+                Type of cpu architecture this ISO is compatible with.
+                None indicates no restriction on the architecture (wildcard).
+            returned: always
```

Comment:
This field is actually not required and can be `null` for `private` isos

Reply:
I hope the wording is correct with the new value `when relevant`.